truffleruby/src/main/ruby/core/truffle/interop.rb

```diff
@@ -63,7 +63,17 @@ def self.enumerable(foreign)
     end
 
     class Foreign
+      def hash
+        Foreign.java_hash(self)
+      end
+      
+      def == b
+        self.object_id == b.object_id
```

Comment:
@chrisseaton I think this is an invalid implementation of `object_id`.
Note `System.identityHashCode` does not have the guarantee of being unique either, at least from JavaDoc.